pilot/pkg/networking/v1alpha3/config.go

```diff
@@ -734,20 +737,33 @@ func buildInboundListeners(mesh *meshconfig.MeshConfig, node model.Proxy,
 // the pod.
 // So, if a user wants to use kubernetes probes with Istio, she should ensure
 // that the health check ports are distinct from the service ports.
-func buildMgmtPortListeners(mesh *meshconfig.MeshConfig, managementPorts model.PortList,
-	managementIP string) []*xdsapi.Listener {
+func buildMgmtPortListeners(managementPorts model.PortList, managementIP string) []*xdsapi.Listener {
 	listeners := make([]*xdsapi.Listener, 0, len(managementPorts))
 
+	if managementIP == "" {
+		managementIP = "127.0.0.1"
+	}
+
 	// assumes that inbound connections/requests are sent to the endpoint address
 	for _, mPort := range managementPorts {
 		switch mPort.Protocol {
 		case model.ProtocolHTTP, model.ProtocolHTTP2, model.ProtocolGRPC, model.ProtocolTCP,
 			model.ProtocolHTTPS, model.ProtocolMongo, model.ProtocolRedis:
-			cluster := v1.BuildInboundCluster(mPort.Port, mPort.Protocol, mesh.ConnectTimeout)
-			listener := buildTCPListener(&v1.TCPRouteConfig{
-				Routes: []*v1.TCPRoute{v1.BuildTCPRoute(cluster, []string{managementIP})},
-			}, managementIP, uint32(mPort.Port), model.ProtocolTCP)
-
+			port := uint32(mPort.Port)
+			config := &tcp_proxy.TcpProxy{
+				StatPrefix: "tcp",
+				// TODO: add tcp routes using deprecated v1 config as filter chain match is incomplete
+			}
+			baseTCPProxy := listener.Filter{
+				Name:   tcpProxyFilter,
+				Config: messageToStruct(config),
+			}
+			listener :=
+				&xdsapi.Listener{
+					Name:         fmt.Sprintf("tcp_%s_%d", managementIP, port),
```

Comment:
helper function for name (TCPListenerName?) would make this more readable. i think this is repeated elsewhere, same for http.

Reply:
Well, just moved it to explicit struct to make it more readable than the shared function...

The problem is that it was not clear what the shared function was doing - a bit of copy seems easier to read.